chore: structure programs

guillaumedebavelaere opened the pull request.

guillaumedebavelaere commented:
I created files suffixed by `escrow`, like `create_escrow`, `withdraw_escrow`, `cancel_escrow`.
I am not sure about this. 
Do you prefer more concise name like just `create`, `withdraw`, `cancel` or to be explicit ?

guillaumedebavelaere commented on anchor/programs/deposit/src/errors.rs:
Not sure if it would be better to be separated in each sub directory ?

guillaumedebavelaere commented on anchor/programs/deposit/src/constants.rs:
Not sure if it would be better to be separated in each sub directory ?

maxnorm replied:
We can keep 1 constant file and adjust if needed. I don't think we will have a lot of them.

guillaumedebavelaere commented on anchor/programs/deposit/src/lib.rs:
```diff
@@ -8,63 +8,6 @@ declare_id!("coUnmi3oBUtwtd9fjeAvSsJssXh5A5xyPbhpewyzRVF");
 pub mod deposit {
```
I find the name of the project not very good because deposit is also an instruction in our programs.
Maybe we should find a more unique and more "marketing" name ?

maxnorm replied:
Trueee. Deposit is straight forward but like you said it basically one of our main functions. It's something we can think while coding

guillaumedebavelaere replied:
@Captain-Cork any thoughts ?

Captain-Cork replied:
Yes, we can change it afterwards. Deposit was a straightforward name though...

maxnorm reviewed (commented):
Maybe adding a enum folder? I'm not sure but it can be useful instead of directly with the states like ModuleType

guillaumedebavelaere commented:
I updated a little bit the structure and added an `accounts` folder. The idea is to make cleaner the state.

`accounts/`: This is where we define how accounts are validated (used in instruction inputs).

`state/`: This is where we define what goes inside on-chain accounts (the data layout of stored accounts).

For example:
`state/user_account.rs`:

```rs
use anchor_lang::prelude::*;

#[account]
pub struct UserAccount {
    pub authority: Pubkey,
    pub username: String,
    pub bump: u8,
}
```

`accounts/create_user.rs`:

```
use anchor_lang::prelude::*;
use crate::state::user_account::UserAccount;

#[derive(Accounts)]
#[instruction(username: String)]
pub struct CreateUser<'info> {
    #[account(
        init,
        payer = authority,
        space = 8 + 32 + 4 + username.len(), // adjust for string length
        seeds = [b"user", authority.key().as_ref()],
        bump
    )]
    pub user: Account<'info, UserAccount>,

    #[account(mut)]
    pub authority: Signer<'info>,

    pub system_program: Program<'info, System>,
}
```
